app/services/guidance_link_service.rb

```diff
@@ -0,0 +1,86 @@
+class GuidanceLinkService
+  class << self
+    def call(document:, chapter: nil, just_the_page_number: false)
+      feature_flag = FeatureFlags.enabled?(:mtr_phase_1, without_session_data: true) ? :mtr_phase_1 : :pre_mtr_phase_1
+
+      if just_the_page_number
+        mapping[feature_flag].dig(document, :sections, chapter)
```

Comment:
Oh, also it would be helpful if these methods raised an error if an unknown chapter was passed in (not sure about 'chapter' by the way, most of these documents aren't divided into chapters, they're divided into sections). So you might considere here and on line 9 using:

```
 mapping[feature_flag].dig(document, :sections).fetch(chapter)
 ```
 
 Fetch will explode if it can't find the specified key, and that will make sure you don't end up with typos causing a bunch of links that don't actually have a page number attached to them.